### PR TITLE
Add tests for a lot of cards that don’t have one yet

### PR DESCRIPTION
**Pull request by blockgoblin31**
Adds tests for Ladder, Lightning Rod, Shield, Wels, Hypno, and Cleo.

### CODE REVIEW

File: tests/game/hermits/welsknight-rare.test.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(
+						TinFoilChefCommon.health -
+							WelsknightRare.secondary.damage -
+							(WelsknightRare.secondary.damage +
+								20) /*extra damage from tfc being in yellow*/ +
```
Comment (Lunarmagpie) [inline tests/game/hermits/welsknight-rare.test.ts:68]: I think this should be welsknight, not TFC. Also the same t hing for the next one.

**Comment by Lunarmagpie** [comment]
I think the Wels ability depends on Wel's health but right now he is not being damaged.

**Comment by Lunarmagpie** [comment]
I switched this pull request to a draft! Please switch it to a regular pull request when you are ready :).

**Comment by blockgoblin31** [comment]
I’m going to go ahead and mark this as ready for review, it has everything I think _needs_ a test and I haven’t worked on it for a couple days so I don’t think I’m going to make more progress. My free time ends next thursday anyway.

**Comment by Lunarmagpie** [comment]
Thank you!

**Review by Screaper91 — APPROVED** [review]

File: common/components/card-component.ts [hunk 2]
```diff
@@ -112,6 +112,10 @@ export class CardComponent<CardType extends Card = Card> {
 			this.game.otherPlayerEntity(this.slot?.player.entity),
 		)
 	}
+	/** Get the type of card */
+	public get type(): CardType {
+		return this.props
+	}
```
Comment (Lunarmagpie) [inline common/components/card-component.ts:118]: Im confused, what is the purpose of this?

Comment (blockgoblin31) [reply common/components/card-component.ts:118]: That's the one thing I wasnt sure how to do so I wrote a method for it, I'm using that in the ladder test to check what hermit is in which slot. I couldnt find anything existing for getting the type of card.

Comment (Lunarmagpie) [reply common/components/card-component.ts:118]: - you should be able to do `expect(CardComponent.props).toBe(TheCardObject)` I think

**Comment by alicetouhou** [comment]
Hey @blockgoblin31 Im going to update this PR's description to be more clear 👍

**Review by Lunarmagpie — APPROVED** [review]
Thanks for the work once again!

**Pull request merged**